diff --git a/rust/private/rustc.bzl b/rust/private/rustc.bzl
index e17f8ac9..fbfe33d 100644
--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -873,6 +873,8 @@ def collect_inputs(
 
     if runtime_libs == None:
         runtime_libs = get_cc_toolchain_runtime_libs(cc_toolchain, feature_configuration, crate_info.type)
+
+    cc_toolchain_all_files = cc_toolchain.all_files if cc_toolchain else depset()
 
     nolinkstamp_compile_inputs = depset(
         nolinkstamp_compile_direct_inputs +
@@ -886,6 +888,7 @@ def collect_inputs(
         ] + ([] if experimental_use_cc_common_link else [
             runtime_libs,
             linker_depset,
+            cc_toolchain_all_files,
         ]),
     )
 
