diff --git a/rust/private/rustc.bzl b/rust/private/rustc.bzl
index e17f8ac9..ad3a6a79 100644
--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -1827,16 +1827,27 @@ def establish_cc_info(ctx, attr, crate_info, toolchain, cc_toolchain, feature_co
         owner = ctx.label,
         libraries = depset([library_to_link]),
     )
+    linker_inputs = [link_input]
+    if crate_info.type in ("staticlib", "cdylib"):
+        if crate_info.type == "cdylib":
+            cc_runtime_libs = cc_toolchain.dynamic_runtime_lib(feature_configuration = feature_configuration)
+        else:
+            cc_runtime_libs = cc_toolchain.static_runtime_lib(feature_configuration = feature_configuration)
+        linker_inputs.append(cc_common.create_linker_input(
+            owner = ctx.label,
+            additional_inputs = depset(transitive = [cc_toolchain.all_files, cc_runtime_libs]),
+        ))
 
     linking_context = cc_common.create_linking_context(
         # TODO - What to do for no_std?
-        linker_inputs = depset([link_input]),
+        linker_inputs = depset(linker_inputs),
     )
 
     cc_infos = [
         CcInfo(linking_context = linking_context),
-        toolchain.stdlib_linkflags,
     ]
+    if crate_info.type != "staticlib":
+        cc_infos.append(toolchain.stdlib_linkflags)
 
     # Flattening is okay since crate_info.deps only records direct deps.
     for dep in crate_info.deps.to_list():
